services/src/main/java/com/chatbot/services/MessageSenders/HangoutsMessageSender.java

```diff
@@ -0,0 +1,100 @@
+package com.chatbot.services.MessageSenders;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.chat.v1.HangoutsChat;
+import com.google.api.services.chat.v1.model.ActionParameter;
+import com.google.api.services.chat.v1.model.Button;
+import com.google.api.services.chat.v1.model.Card;
+import com.google.api.services.chat.v1.model.FormAction;
+import com.google.api.services.chat.v1.model.Message;
+import com.google.api.services.chat.v1.model.OnClick;
+import com.google.api.services.chat.v1.model.Section;
+import com.google.api.services.chat.v1.model.TextButton;
+import com.google.api.services.chat.v1.model.WidgetMarkup;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+// class to handle sending of messages to hangouts users
+
+@Component
+public class HangoutsMessageSender extends MessageSender{
+  private static String CHAT_SCOPE;
+  private static GoogleCredentials credentials;
+  private static HttpRequestInitializer requestInitializer;
+  private static HangoutsChat chatService;
+
+  HangoutsMessageSender(@Value("${hangoutsAPIScope}") final String apiScope,
```

Comment:
ChatService should be inject. Then you could mock it and test.

Reply:
is now loaded as a component on application start up